### ISSUE TITLE
Parallel performance on long running tasks

### ISSUE THREAD

**Issue by tomek-kuchta** [fabric/fabric#1391]
Please correct me if this is not an issue.

```
fab --version
Fabric 1.10.2
Paramiko 1.10.1
```

I was running a command on 8 machines, the task looks as follows:
```
@task
@parallel(pool_size=8)
def test_task():
        run("sleep 60")
```

I'm running it on the VM that is assigned 16 cpus (the physical machine has this number of actual physical cores if we don't count HT).
When this task is run, I see all 16 cpus becoming busy. They are not 100% but the slowdown is noticeable.

Machines that I'm connecting to are LXC containers running under the same VM. 
Thanks

**Comment by tomek-kuchta** [comment]
I've just run the same test but by invoking 8 instances of ```fab```, one for each target, rather than using the ```@parallel``` directive. It seems to me that the outcome is similar (I also see CPU consumption going up).

**Comment by bitprophet** [comment]
There've been some issues re: busy loops & general CPU hogging, but I think some of those were fixed at the Paramiko level; 1.10 is a mite old at this point and I think many/most of those fixes occurred afterwards.

You may want to try upgrading to Paramiko 1.12 (pre-Python3 support, which shook a lot of stuff up, but still a good amount newer than 1.10) or 1.15 (most recent release) and see if that helps.
